### PR TITLE
issue 459.2: ensure view retrieval is denied of detailed definition is denied

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/RestHandler/RestViewHandler.cpp [hunk 1]
```diff
@@ -83,6 +83,25 @@ void RestViewHandler::getView(std::string const& nameOrId, bool detailed) {
     return;
   }
 
+  // skip views for which the full view definition cannot be generated, as per https://github.com/arangodb/backlog/issues/459
+  try {
+    arangodb::velocypack::Builder viewBuilder;
+
+    viewBuilder.openObject();
+
+    auto res = view->toVelocyPack(viewBuilder, true, false);
```
Comment (graetzer) [inline arangod/RestHandler/RestViewHandler.cpp]: isn't there a better way of doing this ?

Comment (graetzer) [reply arangod/RestHandler/RestViewHandler.cpp]: Can't you use `velocypack::Builder::removeLast()` ?

Comment (vasiliy-arangodb) [reply arangod/RestHandler/RestViewHandler.cpp]: I am unsure how removeLast() would be applicable in this situation. According the the description it only applies to the last value and only of an unclosed object, neither of which is guaranteed.

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/789/

**Pull request merged**